blocks/blocks.go

```diff
@@ -124,6 +129,43 @@ func (bh *BlockHandler) HandleBlockData(ctx context.Context, data []byte, fetche
 	return nil
 }
 
+func (bh *BlockHandler) saveMetrics(blk types.Block) {
+	type metric struct {
+		hist  commonMetrics.Histogram
+		value float64
+	}
+
+	metricList := []metric{
+		{
+			hist:  metrics.AvgLayerBlockSize,
+			value: float64(len(blk.Bytes())),
+		},
+		{
+			hist:  metrics.AvgNumTxsInBlock,
+			value: float64(len(blk.TxIDs)),
+		},
+		{
+			hist:  metrics.AvgForDiffLength,
+			value: float64(len(blk.ForDiff)),
+		},
+		{
+			hist:  metrics.AvgNeutralDiffLength,
+			value: float64(len(blk.NeutralDiff)),
+		},
+		{
+			hist:  metrics.AvgAgainstDiffLength,
+			value: float64(len(blk.AgainstDiff)),
+		},
+	}
+
+	for _, m := range metricList {
+		m.hist.
+			With("layer_id", blk.LayerIndex.String()).
```

Comment:
would be better to use constants here and where labels are defined. otherwise this will be prone to typos